src/main/java/com/cactus/systems/GroceryListSystem.java

```diff
@@ -101,46 +33,30 @@ public void logout(){
      * @param name
      * @return true if a new groceryList was created, false otherwise
      */
-    public boolean newGroceryList(String name){
-        GroceryList newGroceryList = this.groceryListManager.createGroceryList(name, this.currentUser.getId());
+    public Response newGroceryList(String name, long userid){
+        Response groceryListResponse = this.groceryListManager.createGroceryList(name, userid);
 
-        if (!Objects.isNull(newGroceryList)){
-            this.currentGroceryList = newGroceryList;
-            return this.repository.saveGroceryList(newGroceryList);
-        } else {
-            return false;
-        }
-    }
+        if (groceryListResponse.getStatusCode() == Status.OK)
+            this.currentGroceryListId = Long.parseLong(groceryListResponse.getPayload().get("listid"));
 
-    /***
-     * Given a category and item name, creates a new item for the current grocery list.
-     * It will return false when the .createItem() function returns a null
-     * telling us that the name was already taken
-     *
-     * @param name
-     * @return true if a new grocery list item was created, false otherwise
-     */
-    public boolean newItem(String name){
-        GroceryItem newGroceryItem = this.groceryListManager.addItemToGroceryList(name, this.currentGroceryList.getId());
-        if(!Objects.isNull(newGroceryItem)){
-            return this.repository.saveGroceryItem(newGroceryItem);
-        } else {
-            return false;
-        }
+        return groceryListResponse;
     }
 
-
     /***
-     * Return a mapping of all the groceryLists from their name to their ID
+     * Return a mapping of all the groceryLists from their ID to their name
      * so that UI can print the names and tell the controller which ID was picked
      *
      * @return groceryListNameMap
      */
-    public HashMap<String, Long> getGroceryListNames(){
-        HashMap<String, Long> groceryListNameMap = new HashMap<String, Long>();
-
-        for(GroceryList groceryList : this.repository.getGroceryListByUser(this.currentUser)){
-            groceryListNameMap.put(groceryList.getName(), groceryList.getId());
+    public HashMap<Long, String> getGroceryListNames(long userid){
```

Comment:
Similar situation here to the above returning Response object situation. I'm not sure the UI necessarily cares about the grocery list IDs, it just cares about the list names to display.

Reply:
For this I thought it would be necessary since when you call getGroceryListNames you are picking a list to edit and so once the ui has to pick from this hash and whichever name they choose they will have to use the id of that list to call getGroceryItems i think

Reply:
I am inclined to agree. I also thought of the fact that if two lists have the same name, then the ID will be needed to distinguish them. I think this we can see if this works, and just make changes as necessary.